### ISSUE TITLE
factory.Faker does not support provider positional arguments

### ISSUE THREAD

**Issue by derekjamescurtis** [FactoryBoy/factory_boy#216]
Certain common providers (such as pylist) require positional arguments.

https://github.com/joke2k/faker/blob/master/faker/providers/python/__init__.py#L46

When additional positional arguments are passed to faker.Factory's constructor, they should be passed along to the fake provider so that I can specify item types for pylist.

https://github.com/rbarrois/factory_boy/blob/master/factory/faker.py#L60

**Comment by rbarrois** [comment]
As stated in #217 , this would make the ``factory.Faker`` API weird, forcing the user to choose the locale to get a list of integers (in your example).

Overall, I have found that mixing ``*args`` and named kwargs fragile, unless using Python 3' "keyword-only arguments" (e.g ``def foo(*args, *, x=3, y=4)``).
If the declaration is ``def foo(x=3, *args)``, users must either use ``foo()`` or ``foo('bar', x=3)``.
The problem comes with library upgrades:
- If the default changes to another value, the user must update all its code accordingly.
- If a new optional kwarg is added, this breaks all existing callers (e.g if the definition is now ``def foo(x=3, y=4, *args)``, the call to ``foo('bar', x=3)`` now reads as ``x=3, y='bar'`` instead of ``x=3, args=['bar']`` as it did in the previous version).

That's why I avoid functions that use both ``*args`` and optional kwargs (``x=3``).
In other words, those signatures are fine:
- ``def foo(x, y, z=3)`` (no ``*args``)
- ``def foo(x, y, *args)`` (no optional parameters)
- ``def foo(x, y, *args, **kwargs)`` (optional parameters might be extracted from ``**kwargs``)
- ``def foo(x, y, z=1, t=2, **kwargs)`` (explicit optional parameters => ``*args`` is forbidden)

**Comment by stefan6419846** [comment]
It seems like this issue has been fixed by *faker* in the meantime and kwargs are supported for these providers now: https://github.com/joke2k/faker/pull/1143
